scratchapi2/user.py

```diff
@@ -1,47 +1,53 @@
-#pylint: disable=trailing-whitespace
 """
 Scratch API User
 
 Anything returned from API does not contain any attributes.
 Use info() to set/get it.
 
+Contains classes:
+* Project
+* User
+* Classroom
 """
-#pylint: enable=trailing-whitespace
 
 import requests
 from .excs import ScratchAPIError
 from .gclass import GenericData
 
+def _request(path, *opts, api_url="https://api.scratch.mit.edu/"):
+    result = requests.get(api_url + path.format(*opts)).json()
+    if 'code' in result:
+        raise ScratchAPIError(result['code'] + ': ' + result['message'])
+
 class Project(object):
-    """ For Scratch Project """
+    """Represents a Scratch Project."""
+
+    projectid = None
+    
     def __init__(self, projectid, getinfo=True):
-        """ Set up Project. """
-        self.projectid = str(projectid)
+        """Initialize a Project."""
+        self.projectid = int(projectid)
         if getinfo:
             self.info()
 
     def __repr__(self):
-        """ Return string <Project (id)> """
+        """Represent a Project."""
         return "<Project {0}>".format(self.projectid)
 
     __str__ = __repr__
 
     def __eq__(self, other):
-        """ Equal means that project IDs are equal """
+        """If the project IDs are equal, the projects are considered equal."""
         return self.projectid == other.projectid
 
     def info(self):
-        """ Get information about a project. It sets its dict and returns it. """
-        try:
-            req = requests.get("https://api.scratch.mit.edu/projects/{0}".format(self.projectid)).json()
-        except:
-            raise ScratchAPIError
-        if "code" in req.keys():
-            raise ScratchAPIError
+        """Get information about a project. It sets its dict and returns it."""
+        req = _request('projects/{0}', self.projectid)
         self.title = req["title"]
         self.description = req["description"]
         self.instructions = req["instructions"]
-        self.author = User(req["author"]["username"], getinfo=False)
+        #this does getinfo because info is only called with getinfo=True anyway
+        self.author = User(req["author"]["username"], getinfo=True)

```

Comment:
The function is only called if getinfo is True; it doesn't hurt to getinfo here too.